workers/unity/Packages/io.improbable.gdk.core/Subscriptions/Systems/SubscriptionSystem.cs

```diff
@@ -12,14 +11,38 @@ public class SubscriptionSystem : ComponentSystem
         private readonly Dictionary<Type, SubscriptionManagerBase> typeToSubscriptionManager =
             new Dictionary<Type, SubscriptionManagerBase>();
 
+        protected override void OnCreate()
+        {
+            base.OnCreate();
+            Enabled = false;
+
+            AutoRegisterManagers();
+        }
+
+        protected override void OnUpdate()
+        {
+        }
+
+        private void AutoRegisterManagers()
+        {
+            var types = ReflectionUtility.GetNonAbstractTypes(typeof(SubscriptionManagerBase),
+                typeof(AutoRegisterSubscriptionManagerAttribute));
+
+            foreach (var type in types)
+            {
+                var instance = (SubscriptionManagerBase) Activator.CreateInstance(type, World);
+                RegisterSubscriptionManager(instance.SubscriptionType, instance);
+            }
+        }
+
         public void RegisterSubscriptionManager(Type type, SubscriptionManagerBase manager)
         {
             if (typeToSubscriptionManager.ContainsKey(type))
             {
                 throw new InvalidOperationException("Already a manager registered");
```

Comment:
can we improve the error messages while we are refactoring it? e.g. add which type of manager is already registered